src/System.Net.Sockets/tests/FunctionalTests/DualModeSocketTest.cs

```diff
@@ -2755,14 +2760,27 @@ protected void ReceiveFrom_Helper(IPAddress listenOn, IPAddress connectTo)
         {
             using (Socket serverSocket = new Socket(SocketType.Dgram, ProtocolType.Udp))
             {
-                serverSocket.ReceiveTimeout = 500;
+                serverSocket.ReceiveTimeout = 1000;
                 int port = serverSocket.BindToAnonymousPort(listenOn);
 
-                SocketUdpClient client = new SocketUdpClient(_log, serverSocket, connectTo, port);
+                SocketUdpClient client = new SocketUdpClient(_log, serverSocket, connectTo, port, sendNow: false);
+
+                if (NetEventSource.IsEnabled)
+                {
+                    NetEventSource.Info(this, $"SRC:{connectTo} {port} DST:{serverSocket.LocalEndPoint}");
+                }
```

Comment:
We generally don't use EventSource / ETW events from tests.

Reply:
I know. That is reason why we get so little information on failures. 
This is direction @karelz  asked me to take. 
With this, we would be able to enable/disable tracing for CI runs.
Possibly have that enabled and collect it on test failures. 

When I debug this I put in custom debug and than I had to remove it. 
Thet seems inefficient. It seems like having test quiet but have ability to be verbose on failure (with useful) details can help us.

Reply:
Then we shouldn't be using NetEventSource, but rather some event source specific to this test library.  And we shouldn't be tracing both to an EventSource and to an ITestOutputHelper, as these tests are doing; we should pick one and be consistent.  Further, at least the output from ITestOutputHelper is grouped with the test output in the test xml file, whereas the EventSource data isn't.

Reply:
ps Did this even work?  I'd expect the answer is "no", as there's no [EventSource] attribute on the NetEventSource, because you're only including the common file rather than one specific to the consuming library that adds the [EventSource] attribute.

Reply:
yes, I can get them if that is your question:
build@toweinfu-ubu14:~/github/corefx/bin/tests/System.Net.Sockets.Tests/netcoreapp-Linux-Debug-x64$ babeltrace test5.trace/lttngTrace/ |grep DualModeConnectionles

[20:01:06.728167800] (+0.000008000) toweinfu-ubu14 DotNETRuntime:EventSource: { packet_seq_num = 263, cpu_id = 0 }, { vpid = 70106, vtid = 70106, procname = "dotnet" }, { EventID = 4, EventName = "Info", EventSourceName = "NetEventSource", Payload = "{\"thisOrContextObject\":\"DualModeConnectionlessReceiveFrom#21083178\", \"memberName\":\"ReceiveFrom_Helper\", \"message\":\"SRC:127.0.0.1 47622 DST:[::]:47622\"}" }
[20:01:06.743739000] (+0.000002200) toweinfu-ubu14 DotNETRuntime:EventSource: { packet_seq_num = 263, cpu_id = 0 }, { vpid = 70106, vtid = 70106, procname = "dotnet" }, { EventID = 4, EventName = "Info", EventSourceName = "NetEventSource", Payload = "{\"thisOrContextObject\":\"DualModeConnectionlessReceiveFrom#21083178\", \"memberName\":\"ReceiveFrom_Helper\", \"message\":\"SRC:127.0.0.1 47622 DST:[::]:47622 RECEIVED:1\"}" }

I'm OK with helper approach or different EventSource. My thinking was that we would like to see other networking events as well for networking tests. 
As it is now, it would be improvement, not necessarily full solution for better debuggability.  I've been also talking to @brianrob about best ways how to utilize tracing.